File: backend/internal/domain/video.go
package domain

import "time"

type Status string

const (
	StatusPending    Status = "pending"
	StatusProcessing Status = "processing"
	StatusReady      Status = "ready"
	StatusError      Status = "error"
)

type Breed string

const (
	BreedSiamese Breed = "siamese" // Bincho
	BreedBengal  Breed = "bengal"  // Kanpachi
	BreedOther   Breed = "other"
)

type Video struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	CatName     string    `json:"catName"`
	Breed       Breed     `json:"breed"`
	Tags        []string  `json:"tags"`
	DurationSec float64   `json:"durationSec"`
	Status      Status    `json:"status"`
	ErrorMsg    string    `json:"errorMsg,omitempty"`
	PlaylistURL string    `json:"playlistUrl,omitempty"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

File: frontend/src/types/video.ts
// Breed slug from the CAT_BREEDS list (lib/breeds.ts). Open string since the
// list can grow without touching this type.
export type Breed = string;

export type VideoStatus = "pending" | "processing" | "ready" | "error";

export interface Video {
  id: string;
  title: string;
  description: string;
  catName: string;
  breed: Breed;
  tags: string[];
  durationSec: number;
  status: VideoStatus;
  errorMsg?: string;
  playlistUrl?: string;
  createdAt: string;
  updatedAt: string;
}
